router/oauthResponseHandler/oauthResponseHandler.go

```diff
@@ -460,7 +460,7 @@ func (authErrHandler *OAuthErrResHandler) cpApiCall(cpReq *ControlPlaneRequestT)
 		reqBody = bytes.NewBufferString(cpReq.Body)
 		req, err = http.NewRequest(cpReq.Method, cpReq.Url, reqBody)
 	} else {
```

Comment:
Naive question. Would there be any issue if we skip this check completely?

Reply:
This is to support `FetchToken` type of calls where in no body is to be sent.

Reply:
That's ok. Would there be an issue sending the empty body as it is?

```
...
var req *http.Request
var err error
// would it fail at this step?
reqBody = bytes.NewBufferString(cpReq.Body)

req, err = http.NewRequest(cpReq.Method, cpReq.Url, reqBody)
if err != nil {
  authErrHandler.logger.Errorf("[%s request] :: destination request failed: %+v\n", loggerNm, err)
  // Abort on receiving an error in request formation
  return http.StatusBadRequest, err.Error()
}
...
```

Reply:
@saikumarrs Want to understand if you feel this if-else is complicated or is there something else that you feel is not right ?! Basically want to understand why the change is being discussed.